### PR TITLE
`deer`: rename `v`, `de` to more meaningful names.

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

This PR is simply cleanup and adds no extra functionality, it renames `v` to `value` in the `Visitor` trait (and it's implementations), or to `object`/`array` for `visit_object` and `visit_array` respectively. It additionally changes `de` to `deserializer`.

Because IDEs often use the default parameter name in the trait as the default name changing the name makes the intent overall clearer.

As a drive-by it renamed the generic of `visit_object` and `visit_array` from `T` (which is very often used for generics in the struct) to `A` (for `A`ccess). This will avoid clashes in auto completion of IDEs and makes everything overall less confusing.

## 🚀 Has this modified a publishable library?

This PR:

- [ ] modifies an **npm**-publishable library and **I have added a changeset file(s)**
- [ ] modifies a **Cargo**-publishable library and **I have amended the version**
- [x] modifies a **Cargo**-publishable library, but **it is not yet ready to publish**
- [ ] modifies a **block** that will need publishing via GitHub action once merged
- [ ] does not modify any publishable blocks or libraries, or modifications do not need publishing
- [ ] I am unsure / need advice

> Looks like this is my 100th PR for HASH. Thank you so much for the continued support and just... everything. Thank you. You guys are awesome 💚 . Until the next milestone 😉

### CODE REVIEW

File: libs/deer/src/lib.rs [hunk 1]
```diff
@@ -195,152 +195,153 @@ pub trait Visitor<'de>: Sized {
...
+    fn visit_string(self, value: String) -> Result<Self::Value, VisitorError> {
+        self.visit_str(&value)
     }
 
-    fn visit_bytes(self, v: &[u8]) -> Result<Self::Value, VisitorError> {
+    fn visit_bytes(self, value: &[u8]) -> Result<Self::Value, VisitorError> {
         Err(Report::new(TypeError.into_error())
             .attach(ReceivedType::new(visitor::BinarySchema::document()))
             .attach(ExpectedType::new(self.expecting()))
             .change_context(VisitorError))
```
Comment (thehabbos007) [inline libs/deer/src/lib.rs:239]: is `value` param unused? can't find it's usage in the fn body

File: libs/deer/src/lib.rs [hunk 2]
```diff
@@ -195,152 +195,153 @@ pub trait Visitor<'de>: Sized {
...
     }
 
-    fn visit_bytes_buffer(self, v: Vec<u8>) -> Result<Self::Value, VisitorError> {
-        self.visit_bytes(&v)
+    fn visit_bytes_buffer(self, value: Vec<u8>) -> Result<Self::Value, VisitorError> {
+        self.visit_bytes(&value)
     }
 
-    fn visit_array<T>(self, v: T) -> Result<Self::Value, VisitorError>
+    fn visit_array<A>(self, array: A) -> Result<Self::Value, VisitorError>
```
Comment (thehabbos007) [inline libs/deer/src/lib.rs:250]: small nit: Just wondering about the consistency between the names here. I see most cases use `value` but array and object are `array` and `object` respectively. Any reason for this distinction?

**Review by thehabbos007 — COMMENTED** [review]
Looks good to me, just have a question for the names

File: libs/deer/src/lib.rs (hunk 1 above)
Comment (indietyp) [reply libs/deer/src/lib.rs:239]: That is correct. This is also why `#[allow(unused_variables)]` is used for this trait. The names are what IDEs suggest for the parameters when filling in the methods, so choosing a name allows us to make auto-complete better, even though we're not using the variable in the default implementation.

File: libs/deer/src/lib.rs (hunk 2 above)
Comment (thehabbos007) [reply libs/deer/src/lib.rs:255]: Gotchu, that's actually an nice distinction, then

**Review by thehabbos007 — APPROVED** [review]
`#100` :tada: :rocket: :rock: :taco: 
Great stuff :)

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**